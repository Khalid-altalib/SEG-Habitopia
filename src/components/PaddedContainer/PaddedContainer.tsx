import React from "react";
import { View } from "react-native";

type Props = {
  children: any;
};

/**
 * Displays children components in a container with predefined padding.
 *
 * @param props The children which are to be displayed within the container.
 * @returns The padded container with its children.
 */
const PaddedContainer = (props: Props) => {
  const PADDING = 15;

  return <View style={{ padding: PADDING }}>{props.children}</View>;
};

export default PaddedContainer;
