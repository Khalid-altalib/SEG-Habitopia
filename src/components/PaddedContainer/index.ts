import PaddedContainer from "./PaddedContainer";

export default PaddedContainer;
