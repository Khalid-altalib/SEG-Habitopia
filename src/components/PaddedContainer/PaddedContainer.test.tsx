// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Habitopia
import PaddedContainer from "@components/PaddedContainer";

it("Renders correctly.", () => {
  render(<PaddedContainer>This is some example content.</PaddedContainer>);

  expect(screen.toJSON()).toMatchSnapshot();
});
