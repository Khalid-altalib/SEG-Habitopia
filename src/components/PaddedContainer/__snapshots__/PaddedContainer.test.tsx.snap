// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<View
  style={
    {
      "height": "100%",
      "padding": 25,
      "width": "100%",
    }
  }
>
  This is some example content.
</View>
`;
